core/scale/optional_bool.cpp

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "scale/optional_bool.hpp"
+
+namespace kagome::scale::optional {
+  template <>
+  outcome::result<void> encodeOptional<bool>(
+      const std::optional<bool> &optional, common::Buffer &out) {
+    uint8_t result = 2;  // true
```

Comment:
Please, make 0, 1 & 2 `static constexpr uint8_t` inside this function, and you will need to comments, as they will be named

Reply:
fixed